### PR TITLE
Remove Cabal dependency from hls-cabal-plugin

### PR DESCRIPTION
**Pull request by cydparser**
Only Cabal-syntax is needed. I was unable to build using Nix without this change due to errors like the following:

```
src/Ide/Plugin/Cabal/Diagnostics.hs:17:1: error:
    Ambiguous module name ‘Distribution.Fields’:
      it was found in multiple packages:
      Cabal-3.6.3.0 Cabal-syntax-3.8.1.0
```

### CODE REVIEW

**Comment by fendor** [comment]
Love the idea, but you will need further adjustments to stack.yaml. I am curious whether this is feasible

**Comment by cydparser** [comment]
@fendor what modifications to stack.yaml are necessary? The stackage resolvers used by this project include Cabal-syntax.

**Comment by fendor** [comment]
If you make it work, great! Last time I checked, it was rather difficult to make it use a different Cabal version. (not sure whether we need a different one)

**Comment by lf-** [comment]
> If you make it work, great! Last time I checked, it was rather difficult to make it use a different Cabal version. (not sure whether we need a different one)

I think this is resolved by use of cabal-syntax and thus cabal version of the toolchain should be possible to ignore, ideally? Needs trying, regardless.

**Comment by lf-** [comment]
I'm trying to build this branch with stack, will let you know how it goes.

**Comment by lf-** [comment]
Aah, shoot, yes, the problem is that Distribution.SPDX.LicenseId is in `Cabal` rather than `Cabal-syntax`. That's unfortunate!

**Comment by lf-** [comment]
```patch
From 61eb593b6406da06cb6d8d2bec09b1c370ccd953 Mon Sep 17 00:00:00 2001
From: Jade Lovelace <jadel@mercury.com>
Date: Mon, 12 Dec 2022 12:29:32 -0800
Subject: [PATCH] Fix the stack build

---
 plugins/hls-cabal-plugin/hls-cabal-plugin.cabal                 | 2 +-
 plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal/LicenseSuggest.hs | 2 +-
 stack-lts19.yaml                                                | 2 +-
 stack.yaml                                                      | 1 +
 4 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/plugins/hls-cabal-plugin/hls-cabal-plugin.cabal b/plugins/hls-cabal-plugin/hls-cabal-plugin.cabal
index eb331283..24ca74bc 100644
--- a/plugins/hls-cabal-plugin/hls-cabal-plugin.cabal
+++ b/plugins/hls-cabal-plugin/hls-cabal-plugin.cabal
@@ -33,7 +33,7 @@ library
   build-depends:
     , base                  >=4.12     && <5
     , bytestring
-    , Cabal-syntax          ^>= 3.6 || ^>= 3.8
+    , Cabal-syntax          ^>= 3.8
     , deepseq
     , directory
     , extra                 >=1.7.4
diff --git a/plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal/LicenseSuggest.hs b/plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal/LicenseSuggest.hs
index 6165cfd1..2236444b 100644
--- a/plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal/LicenseSuggest.hs
+++ b/plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal/LicenseSuggest.hs
@@ -24,7 +24,7 @@ import           Language.LSP.Types          (CodeAction (CodeAction),
 import           Text.Regex.TDFA
 
 import qualified Data.List                   as List
-import           Distribution.SPDX.LicenseId (licenseId)
+import           Distribution.SPDX           (licenseId)
 import qualified Text.Fuzzy.Parallel         as Fuzzy
 
 -- | Given a diagnostic returned by 'Ide.Plugin.Cabal.Diag.errorDiagnostic',
diff --git a/stack-lts19.yaml b/stack-lts19.yaml
index 74d90c33..d21deedd 100644
--- a/stack-lts19.yaml
+++ b/stack-lts19.yaml
@@ -42,7 +42,7 @@ ghc-options:
   "$everything": -haddock
 
 extra-deps:
-- Cabal-3.6.0.0
+- Cabal-syntax-3.8.0.0
 # needed for tests of hls-cabal-fmt-plugin
 - cabal-fmt-0.1.6@sha256:54041d50c8148c32d1e0a67aef7edeebac50ae33571bef22312f6815908eac19,3626
 - floskell-0.10.6@sha256:e77d194189e8540abe2ace2c7cb8efafc747ca35881a2fefcbd2d40a1292e036,3819
diff --git a/stack.yaml b/stack.yaml
index b9244827..7863d00d 100644
--- a/stack.yaml
+++ b/stack.yaml
@@ -52,6 +52,7 @@ extra-deps:
 - lsp-types-1.6.0.0
 - lsp-test-0.14.1.0
 - hie-bios-0.11.0
+- Cabal-syntax-3.8.1.0
 
 # currently needed for ghcide>extra, etc.
 allow-newer: true
-- 
2.38.1
```

This patch fixes the stack build on your branch. I would push to your branch but don't have write access (don't want to accidentally become a HLS maintainer)

**Comment by lf-** [comment]
> stackage-lts19 failure

This is due to the removal of the `cabal-3.6.0.0` line, which is apparently required by hls-ormolu-plugin somehow? I thought it wasn't necessary so deleted it after testing but I was mistaken.

> stackage-nightly failure

Fixed in https://github.com/commercialhaskell/stack/pull/5886

**Review by fendor — APPROVED** [review]
LGTM, if stack CI is green :)

**Comment by cydparser** [comment]
Cabal-syntax is no longer a dependency: https://github.com/haskell/haskell-language-server/commit/f5db1507d40f393b01ef2c2b817f819aa2ffcc7a#diff-5e8d34349db491632161012cabc6a29aa032853e30534abb7e36d9462a0b1df2L48

**Pull request closed without merging**
